src/AutoRest.CSharp/Common/Output/Models/Types/HelperTypeProviders/MultipartFormDataRequestContentProvider.cs

```diff
@@ -0,0 +1,495 @@
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using AutoRest.CSharp.Common.Input;
+using AutoRest.CSharp.Common.Output.Expressions.KnownValueExpressions;
+using AutoRest.CSharp.Common.Output.Expressions.Statements;
+using AutoRest.CSharp.Common.Output.Expressions.ValueExpressions;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Generation.Writers;
+using AutoRest.CSharp.Output.Models;
+using AutoRest.CSharp.Output.Models.Serialization;
+using AutoRest.CSharp.Output.Models.Shared;
+using AutoRest.CSharp.Output.Models.Types;
+using static AutoRest.CSharp.Common.Output.Models.Snippets;
+using MultipartFormDataContent = System.Net.Http.MultipartFormDataContent;
+
+namespace AutoRest.CSharp.Common.Output.Models.Types.HelperTypeProviders
+{
+    internal class MultipartFormDataRequestContentProvider: ExpressionTypeProvider
+    {
+        private static readonly Lazy<MultipartFormDataRequestContentProvider> _instance = new(() => new MultipartFormDataRequestContentProvider());
+        private class FormDataItemTemplate<T> { }
+        public static MultipartFormDataRequestContentProvider Instance => _instance.Value;
+        private MultipartFormDataRequestContentProvider() : base(Configuration.HelperNamespace, null)
+        {
+            DeclarationModifiers = TypeSignatureModifiers.Internal;
+            Inherits = Configuration.ApiTypes.RequestContentType;
+            _multipartContentField = new FieldDeclaration(
+                modifiers: FieldModifiers.Private | FieldModifiers.ReadOnly,
+                type: typeof(MultipartFormDataContent),
+                name: "_multipartContent");
+            _randomField = new FieldDeclaration(
+                modifiers: FieldModifiers.Private | FieldModifiers.Static | FieldModifiers.ReadOnly,
+                type: typeof(Random),
+                name: "_random",
+                initializationValue: New.Instance(typeof(Random), Array.Empty<ValueExpression>()),
+                serializationFormat: SerializationFormat.Default);
+            _boundaryValuesFields = new FieldDeclaration(
+                modifiers: FieldModifiers.Private | FieldModifiers.Static | FieldModifiers.ReadOnly,
+                type: typeof(char[]),
+                name: "_boundaryValues",
+                initializationValue: Literal("0123456789=ABCDEFGHIJKLMNOPQRSTUVWXYZ_abcdefghijklmnopqrstuvwxyz").ToCharArray(),
+                serializationFormat: SerializationFormat.Default);
+            ValueExpression contentTypeHeaderExpression = new UnaryOperatorExpression("!", new MultipartFormDataContentExpression(_multipartContentField).Property("Headers").Property("ContentType"), true);
```

Comment:
Please add the `Headers` and `ContentType` property in the corresponding expression types, so that we could have:
```suggestion
            ValueExpression contentTypeHeaderExpression = new UnaryOperatorExpression("!", new MultipartFormDataContentExpression(_multipartContentField).Headers.ContentType, true);
```

Reply:
By checking your generated code, we are wanting `_multipartContent.Headers.ContentType!`.
The whole idea of expressions is to make everything to have the correct csharp semantic meaning.
Here, this `!` is the null forgiving operator which is not a `UnaryOperatorExpression`. We should not constructing this here.
Instead, since we did not enable nullable context in all our generated code, we do not need to add the `!` operator here.
Please just change this to:
```suggestion
            ValueExpression contentTypeHeaderExpression = new MultipartFormDataContentExpression(_multipartContentField).Headers.ContentType, true);
```

Reply:
`ContentType` is property of `System.Net.Http.Headers.HttpContentHeaders`  and the contentType expression only used here, so Maybe we will not add a TypedValueExpression for `System.Net.Http.Headers.HttpContentHeaders`. 

`Headers` will update.